### PR TITLE
Create de Vries scorer, support wrapping phenotype scoring function

### LINKED ISSUE #217: De Vries Score
De Vries Score

This is an example of a bespoke severity score.
It would be a stretch goal to define an interface to allow the user to specify models like this dynamically, but for now, let's define a predicate that does this, comparable to the existing predicate for counting up features (that is dynamic and takes a set of HPO terms as input - that would not work here).

Defined in PMID: 21712853 as the adapted De Vries Score
The De Vries score was developed as a relatively simple phenotypic severity score for individuals with intellectual disability in which points are given for (severity of) intellectual disability, growth abnormalities (prenatal and postnatal), facial dysmorphisms, non-facial dysmorphisms, and other congenital anomalies
 
Points are given according to the following system
Developmental delay
Mild developmental delay 1
Moderate developmental delay 1
Severe developmental delay 2
Postnatal growth abnormality (Max 2)
Microcephaly (1)
+Short stature (1)
Macrocephaly (1)
>=2 facial dysmorphic features: 2
Non-facial dysmorphism and congenital abnormalities (hand anomaly, heart anomaly, hypospadias with/without undescended testis; assign 1 point for each, with a maximum score of 2 points)
 
The maximum number of points is 10.

### PR DESCRIPTION
**Pull request by ielis**
@pnrobinson here I show a way to wrap a phenotype scoring function/closure and I add a stub for de Vries score.

Please implement the `gpsea.analysis.pscore._impl.DeVriesPhenotypeScore` or let me know if I or Lauren should do it.

Thanks!

Fixes #217

### CODE REVIEW

**Comment by pnrobinson** [comment]
@ielis We are getting errors like this with the CI
```
requests.exceptions.HTTPError: 429 Client Error: TOO MANY REQUESTS for url: https://rest.variantvalidator.org/VariantValidator/tools/gene2transcripts/NM_181486.4
````
but I think things are working. Maybe there is some way to add a sleep(500 ms) between the three rounds of CI?

File: src/gpsea/analysis/pscore/_impl.py [hunk 1]
```diff
@@ -110,3 +110,178 @@ def score(
...
+        Args:
+            observed_term_ids: terms observed in patient
+
+        Returns: facial dysmorphism score (between 0 and 2)
+
+        """
+        hypertelorism = 'HP:0000316'
+        external_nose = 'HP:0010938'
+        pinna_morphology = 'HP:0000377'
+        total_count = len([t for t in observed_term_ids if t == hypertelorism])
```
Comment (ielis) [inline src/gpsea/analysis/pscore/_impl.py]: There is a conflict between the pydoc and the actual code - the code actually does not include the Hypertelorism descendants.

I will update the code to include the descendants.

File: src/gpsea/analysis/pscore/_impl.py [hunk 2]
```diff
@@ -110,3 +110,178 @@ def score(
...
+        Args:
+            observed_term_ids:  terms observed in patient
+
+        Returns:   Non-facial dysmorphism and congenital abnormalities score (between 0 and 2)
+
+        """
+        hypospadias = 'HP:0000047'
+        abnormal_hand_morphology = 'HP:0005922'
+        abnormal_heart_morphology = 'HP:0001627'
+        total_count = len([t for t in observed_term_ids if t == hypospadias])
```
Comment (ielis) [inline src/gpsea/analysis/pscore/_impl.py]: Same as above, this code does not check Hypospadias descendants.

I will fix the code to check the descendants. @pnrobinson please should if it should not be done!

**Review by ielis — COMMENTED** [review]
The code looks great, I love the documentation and tests. There are few things below, however, that I would like to double check with you, because I think the pydoc does not correspond to code.

Can you pls check the comments?

**Review by ielis — COMMENTED** [review]
Regarding the failing CI, in the short term we need to re-run the jobs. In the long term, we will update the suite to ensure we do not in fact need to reach out to the network.

So, I think it is OK as it is.

**Review by ielis — COMMENTED** [review]
Regarding the documentation, we will need to shuffle the sections. That should be, however, easy. Let's discuss in person.

File: src/gpsea/analysis/pscore/_impl.py (hunk 1 above)
Comment (pnrobinson) [reply src/gpsea/analysis/pscore/_impl.py]: Hypertelorism does not have descendents and so this implementation is a bit quicker

File: src/gpsea/analysis/pscore/_impl.py (hunk 2 above)
Comment (pnrobinson) [reply src/gpsea/analysis/pscore/_impl.py]: see above, hypertelorism does not have descendats

File: src/gpsea/analysis/pscore/_impl.py [hunk 3]
```diff
@@ -110,3 +110,178 @@ def score(
...
+    def _term_or_descendant_count(self,
+                                  target_tid: str,
+                                  observed_term_ids: typing.List[str]):
+        """
+        Args:
+            target_tid: term of interest
+            observed_term_ids: all terms observed in patient
+
+        Returns:
+            1 if the term or any descendant is present in the patient, otherwise 0
```
[the remark this answers is not in the document]
Comment (pnrobinson) [reply src/gpsea/analysis/pscore/_impl.py:186]: Sorry, the documentation was wrong. This returns the total count of term or descendants!

File: src/gpsea/analysis/pscore/_impl.py (hunk 2 above)
Comment (ielis) [reply src/gpsea/analysis/pscore/_impl.py]: I am afraid I got you on this one.. https://hpo.jax.org/browse/term/HP:0000047

Hypospadias actually *does* have descendants.

File: src/gpsea/analysis/pscore/_impl.py (hunk 1 above)
Comment (ielis) [reply src/gpsea/analysis/pscore/_impl.py]: OK, I'll revert.

**Comment by ielis** [comment]
@pnrobinson I think this looks good. Pls merge if you do not plan any other extensions.

**Comment by pnrobinson** [comment]
Yeah then we should include the descendants!

**Review by pnrobinson — APPROVED** [review]
Looks good with changes, I will merge!

**Pull request merged**